apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -1,14 +1,18 @@
 use serialport::{self, SerialPort};
-use std::{fmt::Display, io, thread, time::Duration};
+use std::{
+    io, thread,
+    time::{Duration, Instant},
+};
 use uinput::{event::keyboard, Device};
 
-const POLL_INTERVAL_MS: u64 = 10;
+const POLL_INTERVAL: Duration = Duration::from_millis(10);
+const MAX_ECHO_RESPONSE_WAIT: Duration = Duration::from_secs(1);
 const UINPUT_PATH: &str = "/dev/uinput";
 const DEVICE_PATH: &str = "/dev/ttyACM1";
 const DEVICE_BAUD_RATE: u32 = 9600;
-const EXPECTED_PACKET_LEGNTH: usize = 7;
+const EXPECTED_PACKET_LENGTH: usize = 7;
 
-#[derive(Debug, PartialEq, thiserror::Error)]
```

Comment:
This `PartialEq` was only needed for the `assert_eq!` in tests, and was causing problems with wrapping `uinput::Error`, so I removed it and used different methods in the tests.